server/src/main/java/com/flipkart/varadhi/CoreServices.java

```diff
@@ -81,20 +80,43 @@ public MeterRegistry getMetricsRegistry() {
        - Also it should be injected dynamically.
      */
 
-    private PersistenceProvider getPersistenceProvider(DBOptions DBOptions) {
-        PersistenceProvider provider = new ZookeeperProvider();
-        provider.init(DBOptions);
-        return provider;
+    private MetaStoreProvider getMetaStoreProvider(MetaStoreOptions metaStoreOptions) {
+        try {
+            String className = metaStoreOptions.getProviderClassName();
+            if (null != className && !className.isBlank()) {
+                Class<MetaStoreProvider> clazz = (Class<MetaStoreProvider>)Class.forName(className);
+                MetaStoreProvider provider = clazz.getDeclaredConstructor().newInstance();
+                provider.init(metaStoreOptions);
+                return provider;
+            }
+            throw new InvalidConfigException();
+        }catch(Exception e) {
+            String errorMsg = String.format("MetaStoreProvider(%s) load failure.", metaStoreOptions.getProviderClassName());
+            log.error(errorMsg, e);
+            throw new InvalidConfigException(errorMsg, e);
+        }
```

Comment:
Extract the common code to some common place?

Reply:
done